### PR TITLE
fix(DataTable): fix cell size and border in FF

### PR DESCRIPTION
**Pull request by booc0mtaco**
- add classes for (header) cell containers
- fix cell height size
- tune up tests, stories, and snapshots

### Test Plan:

- [x] Wrote/updated [automated tests](https://czi.atlassian.net/wiki/x/Hbl1H)
- [ ] CI tests / new tests are not applicable
- [ ] Manually tested my changes, and here are the details:

### CODE REVIEW

**Comment by booc0mtaco** [comment]
There is one remaining issue noticed here, with the right-hand border in the right-most header cell. Will address in a separate PR, with any other firefox discrepancies

File: src/components/DataTable/DataTable.tsx [hunk 1]
```diff
@@ -213,6 +213,7 @@ export function DataTable<T>({
               <DataTableRow key={headerGroup.id}>
                 {headerGroup.headers.map((header) => (
                   <th
+                    className={styles['data-table__header-cell-container']}
```
Comment (sskikne) [inline src/components/DataTable/DataTable.tsx:216]: Not a complaint, but curious about this naming format that mizes underscore and dash

**Review by sskikne — APPROVED** [review]
lgtm!

File: src/components/DataTable/DataTable.tsx (hunk 1 above)
Comment (booc0mtaco) [reply src/components/DataTable/DataTable.tsx:216]: Yep yep, [doc.s here](https://chanzuckerberg.github.io/edu-design-system/?path=/docs/documentation-contributor-guidelines-code-guidelines--docs)

In our scheme, `-` has no special meaning and is in place of using camelCase in CSS naming, preferring snake-case. `--` and `__` have more strict meaning, with most of the rest is convention

HOLD ME ACCOUNTABLE to that page :)

**Comment by booc0mtaco** [comment]
thanks @sskikne ✅

**Pull request merged**